### DISCUSSION TITLE
JMX Port refused to connect

### DISCUSSION THREAD

**Discussion by raghulp18** [strimzi/strimzi-kafka-operator#10213] [General]
Hello Strimzi Community,

I have enabled both jmxoptions and kafka connect metrics in strmzi kafka connect, the kafka connect metrics are exposed on port 9404 under path /metrics/. But I'm not able to connect to port 9999 where jmx metrics are exposed.  

kafka connect pod description:

![image](https://github.com/strimzi/strimzi-kafka-operator/assets/94948561/2615df69-d4ac-44c6-9a72-72a6c7085b05)

Connection refused at port 9999

<img width="953" alt="image" src="https://github.com/strimzi/strimzi-kafka-operator/assets/94948561/1c36add3-a740-40e2-b376-e92b34e2e793">

**Comment by im-konge** [comment]
The JMX stuff can be accessed by JMX terminal (for example from here https://github.com/jiaqi/jmxterm/releases) - at least that's what I used for testing the JMX metrics.
IIRC I wasn't able to do the curl commands there.

**Reply by scholzj** [reply]
JMX is not HTTP based, so no, you will never be able to use it with `curl`.

**Reply by raghulp18** [reply]
Ok Thanks for the quick response. Will I able to run jconsole inside a kafkaconnect pod ? How can i scrap out the kafka connect metrics (which running on 9404/metrics/ ) outside the strimzi cluster and expose it to prometheus and grafana ?

**Reply by scholzj** [reply]
JMX has nothing to do with Prometheus metrics. If you want to use Prometheus metrics, scrape them from port 9404 as we discussed in https://github.com/orgs/strimzi/discussions/10198.

**Reply by raghulp18** [reply]
@im-konge  If i want to use jmx terminal which you shared, where do i have place the jar inside strimzi kafka connect, whether it is under opt/kafka/libs ?? or somewhere else ? Can you please share the path ?

**Reply by im-konge** [reply]
you don't need to place it anywhere specifically, I think that you will be able to have it in /tmp for example or somewhere else. It doesn't need to be somewhere specifically.

**Reply by raghulp18** [reply]
Agreed @scholzj. I got the metrics at 9404 but how do i scrape them from 9404, that's my question. And how do I expose it in prometheus UI or grafana instead of the just seeing the metrics inside a kafka connect pod ?

**Reply by im-konge** [reply]
Everything around metrics and how to expose them for Prometheus and Grafana is mentioned in our documentation, so I would start there

**Reply by scholzj** [reply]
You have to configure your Prometheus to scrape them. The exact details are up to how you manage, install and configure your Prometheus.

**Reply by raghulp18** [reply]
Can you provide any example files to configure ?

**Reply by scholzj** [reply]
As @im-konge said, I think there are some examples in our docs and examples. But in general, this is up to you. Configuring and managing Prometheus is not our area of expertise.
